### PR TITLE
Feature/preview

### PR DESCRIPTION
**Pull request by VladMartyshevskyi**
Integrated editor button to ECMS preview

### CODE REVIEW

File: services/src/main/java/org/exoplatform/officeonline/documents/OfficeOnlineDocumentEditorPlugin.java [hunk 1]
```diff
@@ -102,39 +129,75 @@ public void initActivity(String uuid, String workspace, String activityId) throw
...
-  public void initPreview(String uuid, String workspace, String activityId, String context, int index) throws Exception {
-    Node symlink = wopiService.nodeByUUID(uuid, workspace);
-    Node node = wopiService.getNode(symlink.getSession().getWorkspace().getName(), symlink.getPath());
-    if (node != null) {
-      if (symlink.isNodeType("exo:symlink")) {
-        String userId = WebuiRequestContext.getCurrentInstance().getRemoteUser();
-        wopiService.addFilePreferences(node, userId, symlink.getPath());
+  public EditorSetting initPreview(String fileId, String workspace, URI requestURI, Locale locale) {
+    try {
+      String userId = ConversationState.getCurrent().getIdentity().getUserId();
```
Comment (pnedonosko) [inline services/src/main/java/org/exoplatform/officeonline/documents/OfficeOnlineDocumentEditorPlugin.java]: Aquire this only where you actually consume it, below at line 160.

File: webapp/src/main/webapp/js/officeonline.js [hunk 2]
```diff
@@ -105,7 +106,8 @@
       var viewerSrc = $vieverScript.attr("src");
       $vieverScript.remove();
       $(".document-preview-content-file").append("<script src='" + viewerSrc + "'></script>");
-    }, 250); // XXX we need wait for office preview server generate a new preview
+    }, 250); // XXX we need wait for office preview server generate a new
```
Comment (pnedonosko) [inline webapp/src/main/webapp/js/officeonline.js]: now very nice comment formatting (multiline), can make shorten or move?

File: webapp/src/main/webapp/js/officeonline.js [hunk 3]
```diff
@@ -346,7 +349,34 @@
       return $("<li class='hidden-tabletL'><a href='" + editorLink + "' target='_blank'>"
           + "<i class='uiIconEcmsOfficeOnlineOpen uiIconEcmsLightGray " + iconClass + "'></i>" + label + "</a></li>");
 
-    }
+    };
+
+    var init = function(userId, cometdConf, userMessages) {
+      if (userId == currentUserId) {
```
Comment (pnedonosko) [inline webapp/src/main/webapp/js/officeonline.js]: Strict equality (`===`) would do a better job, agree?

File: webapp/src/main/webapp/WEB-INF/conf/officeonline/configuration.xml [hunk 4]
```diff
@@ -591,11 +591,11 @@
             </field>
             <field name="patterns">
               <collection type="java.util.ArrayList" item-type="java.lang.String">
-                <value>
-                  <string>/intranet/documents.*</string>
+               <value>
+                  <string>.*/documents.*</string>
```
Comment (pnedonosko) [inline webapp/src/main/webapp/WEB-INF/conf/officeonline/configuration.xml:595]: In this case we'll work for all possible prefixes what is more than we want to do. As it's a regexp, we could restrict for only `/intranet` or `/dw` in all such places.

**Review by pnedonosko — CHANGES_REQUESTED** [review]

File: services/src/main/java/org/exoplatform/officeonline/documents/OfficeOnlineDocumentEditorPlugin.java (hunk 1 above)
Comment (VladMartyshevskyi) [reply services/src/main/java/org/exoplatform/officeonline/documents/OfficeOnlineDocumentEditorPlugin.java]: Fixed: https://github.com/exo-addons/office-online/pull/30/commits/2495d2de3e4c5cd7f2fa6dd97275a06344d55e73

File: webapp/src/main/webapp/js/officeonline.js (hunk 2 above)
Comment (VladMartyshevskyi) [reply webapp/src/main/webapp/js/officeonline.js]: Fixed: https://github.com/exo-addons/office-online/pull/30/commits/2495d2de3e4c5cd7f2fa6dd97275a06344d55e73

File: webapp/src/main/webapp/js/officeonline.js (hunk 3 above)
Comment (VladMartyshevskyi) [reply webapp/src/main/webapp/js/officeonline.js]: Fixed: https://github.com/exo-addons/office-online/pull/30/commits/2495d2de3e4c5cd7f2fa6dd97275a06344d55e73

File: webapp/src/main/webapp/WEB-INF/conf/officeonline/configuration.xml (hunk 4 above)
Comment (VladMartyshevskyi) [reply webapp/src/main/webapp/WEB-INF/conf/officeonline/configuration.xml:595]: This filter will be deleted in following PR https://github.com/exo-addons/office-online/pull/31

**Review by pnedonosko — APPROVED** [review]

**Pull request merged**